Added support for TileJson files & for UTFgrid tiles

infostreams opened the pull request:
Hi Zverik,

I refactored your code and added support for TileJSON files and for UTFgrid files. That way it can now also serve the interactivity layer of maps produced with TileMill. I used your code as a basis, but ended up extending & changing it so much that it almost looks unrecognizable. Your code is still there though, just transmogrified ;-)

In any case, I thought I'd let you know. Care to pull?

infostreams committed:
- Refactored
- Added UTFGrid & TileJSON support
- Added routing logic

infostreams committed:
Moved routing logic to .php file

infostreams committed:
Added some additional information

infostreams committed:
Renamed it to tileserver.php

infostreams committed:
Formatted code, slight polishing

infostreams committed:
Formatted code (now for real)

infostreams committed:
Never trust a machine to format your source code... (sigh)

infostreams committed:
Formatting

infostreams committed:
File header

infostreams committed:
- Updated code that serves UTFgrid tiles
- SQlite connection is no longer persistent

infostreams committed:
Ignore .mbtiles files

infostreams committed:
- in the TileJson file, make sure that maxzoom & minzoom are integers, not strings
- also, make sure that 'center' and 'bounds' are floats, not strings

infostreams committed:
Added a sentence & re-formatted for BitBucket

kurt-hectic committed:
fixed bug that let warning about undefined id parameter come up on picky systems

Zverik commented:
Hi! This is really grand update! Thanks for all the work! I think of not merging this pull request, but recommending to use your fork instead of mine in readme.

Also, I've installed your script, but it gives "Page no found" for whatever links I'm throwing at it. There is a complex nginx+apache stack, so http://localhost/mbtiles/root.xml becomes /~zverik/mbtiles/root.xml which is processed with mod_rewrite.

infostreams committed:
Added sane browser caching (with Expires headers, ETags, and HTTP 304 status codes)

infostreams committed:
Merge branch 'master' of bitbucket.org:wmo/cpdb-mapserver

infostreams committed:
Added $tileset member to prevent PHP notice when generating ETAG

infostreams committed:
Include the 'modified date' of the .MBTiles file in the ETag header

cwhite911 committed:
Added directions for implementing ubuntu aws server configured to run mbtiles-php

cwhite911 committed:
Created new file called server-setup

cwhite911 committed:
Update server-setup.md

Changed format

infostreams committed:
Merge pull request #1 from cwhite911/master

server setup instructions

infostreams committed:
Recognize if we're running on HTTPS domain and act accordingly

infostreams committed:
Merge branch 'master' of github.com:infostreams/mbtiles-php

Fil committed:
allow serving static files from the tiles/ directory

if static files (and sub-directories) are present, don't route through tileserver.php
this allows:
1. to switch from a static tree to an .mbtiles, and back, without changing URLs
2. to roughly build a cache for (a subset of) the tiles

infostreams committed:
Merge pull request #3 from Fil/allow-static-files

allow serving static files from the tiles/ directory

helmi03 committed:
fix rewrite and sample file

infostreams committed:
What the....

infostreams committed:
Don't ignore directories

infostreams committed:
Merge branch 'master' of github.com:infostreams/mbtiles-php

infostreams committed:
Renamed

infostreams committed:
Added example

infostreams committed:
Added mbtiles basemap

infostreams committed:
Match helmi03's version so he gets to keep the credits for the following commit...

infostreams committed:
Back to previous version

infostreams committed:
Fix for deploying in domain root

Courtesy helmi03

infostreams committed:
Partially merged with 'index.html'

infostreams committed:
Updated example to use mapbox.js

infostreams committed:
Added example to showcase map with layers

infostreams committed:
New version that includes Caspian Sea ;-)

infostreams committed:
New example maps

klokan commented:
BTW There is now a ready to use PHP server for MBTiles + TileJSON + UTFGrid hosting available from: https://github.com/klokantech/tileserver-php/

with a live demo at:
http://tileserver.maptiler.com/

![68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f35393238342f313034313830372f61303430313630632d306664622d313165332d383934312d6162333637623261363438642e706e67](https://cloud.githubusercontent.com/assets/59284/3102295/fd777726-e644-11e3-9022-0bd9957dad36.png)

So you can try that one as well...

infostreams committed:
Fixes for GitHub issues #11 and #12

infostreams committed:
Fix for Github issue #13

infostreams committed:
Updated with Nginx installation instructions & system requirements (thanks tbleiker!)

infostreams committed:
Fix for issue #15 - support for Retina maps

infostreams committed:
Formatting

infostreams committed:
Added section on Retina maps

infostreams committed:
Typo

infostreams committed:
Adjusted routing code to better deal with symlinks

infostreams committed:
Added some error detection

infostreams committed:
Textual changes, bumped version to 1.3

infostreams committed:
Replaced with symlink
